src/overlays/actors/ovl_En_Kame/z_en_kame.c

```diff
@@ -182,254 +211,329 @@ void func_80AD7018(EnKame* this, PlayState* play) {
+ */
+void EnKame_RetreatIntoShell(EnKame* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
     if (SkelAnime_Update(&this->snapperSkelAnime)) {
         if ((Player_GetMask(play) != PLAYER_MASK_STONE) && !(player->stateFlags1 & PLAYER_STATE1_800000) &&
-            ((this->unk_29E == 0) || (this->actor.xzDistToPlayer < 120.0f))) {
-            func_80AD76CC(this);
+            ((this->timer == 0) || (this->actor.xzDistToPlayer < 120.0f))) {
+            EnKame_SetupPrepareToAttack(this);
         } else {
-            this->unk_29E--;
-            if (this->unk_29E == 0) {
-                func_80AD7B18(this);
+            this->timer--;
+            if (this->timer == 0) {
+                EnKame_SetupEmergeFromShell(this);
             }
         }
     } else if (this->snapperSkelAnime.curFrame > 2.0f) {
-        this->unk_2AC = 1.5f - ((this->snapperSkelAnime.curFrame - 2.0f) * (7.0f / 30));
-        this->unk_2A8 = 1.5f - ((this->snapperSkelAnime.curFrame - 2.0f) * (1.0f / 12));
+        this->limbScaleY = 1.5f - ((this->snapperSkelAnime.curFrame - 2.0f) * (7.0f / 30));
+        this->limbScaleXZ = 1.5f - ((this->snapperSkelAnime.curFrame - 2.0f) * (1.0f / 12));
     } else {
-        f32 frame = this->snapperSkelAnime.curFrame;
+        f32 curFrame = this->snapperSkelAnime.curFrame;
 
-        this->unk_2AC = (0.25f * frame) + 1.0f;
-        this->unk_2A8 = (0.25f * frame) + 1.0f;
+        this->limbScaleY = (0.25f * curFrame) + 1.0f;
+        this->limbScaleXZ = (0.25f * curFrame) + 1.0f;
     }
 }
 
-void func_80AD7568(EnKame* this) {
-    this->actor.speed = this->unk_2A6 * (5.0f / 7552);
-    this->actor.shape.rot.z = this->unk_2A6 * 0.11016949f;
+/**
+ * Sets the Snapper's speed and z-rotation based on its angular velocity. When the Snapper is done with the first
+ * part of its attack, it will slowly "wind up" before moving back to its home; by setting the speed to be a multiple
+ * of its angular velocity, the Snapper will smootly ramp up its speed during the "wind up". As for the z-rotation,
+ * setting it in this manner gives the Snapper a subtle, almost-imperceptible tilt as it spins.
+ */
+void EnKame_SetAttackSpeed(EnKame* this) {
+    this->actor.speed = this->angularVelocityY * (5.0f / 0x1D80);
+    this->actor.shape.rot.z = this->angularVelocityY * (13.0f / 118.0f);
 }
 
-void func_80AD75A8(EnKame* this, PlayState* play) {
-    static Color_RGBA8 D_80AD8E54 = { 250, 250, 250, 255 };
-    static Color_RGBA8 D_80AD8E58 = { 180, 180, 180, 255 };
-    static Vec3f D_80AD8E5C = { 0.0f, 0.75f, 0.0f };
+/**
+ * If the Snapper is moving quickly enough over certain surface materials, this function will spawn dirt or snow-colored
+ * dust on even-numbered frames. Additionally, this function is also responsible for playing the Snapper's spinning
+ * sound effect if its angular velocity is large enough.
+ */
+void EnKame_PlayAttackEffects(EnKame* this, PlayState* play) {
```

Comment:
`EnKame_PlayAttackEffects` is the only name that sounds a bit odd to me. I think it's just the `Play` being used on `AttackEffects`, when this function spawns surface effects and plays the spinning sfx. Maybe `EnKame_ProcessAttackEffects`?

Reply:
Yeah, that's fine. I didn't quite know what to call this because it has two different responsibilities really.